src/app/authentication/login/login.service.ts

```diff
@@ -14,10 +14,12 @@ export class LoginService {
   login(login: Login): Observable<any> {
     // tslint:disable-next-line:max-line-length
     const body = `username=${login.username}&password=${login.password}&client_id=${environment.OAUTH_CLIENT_ID}&client_secret=${environment.OAUTH_CLIENT_SECRET}&grant_type=password`;
+    const clientCreds = `${environment.OAUTH_CLIENT_ID}:${environment.OAUTH_CLIENT_SECRET}`;
 
     let headers = new HttpHeaders();
-    headers = headers.set('grant_type', 'password');
-    headers = headers.set('content-type', 'application/x-www-form-urlencoded');
+
+    headers = headers.set('Content-Type', 'application/x-www-form-urlencoded');
```

Comment:
Removing the grant type header doesn't work with the .NET side, but that's just because we check for it when validating the client. I can change that.

Reply:
The java side can also ignore that header, but I would vote for the clean up in .net api if it isn't needed.  I couldn't tell when reading the OAuth RFC https://tools.ietf.org/html/rfc6749 if it was optional to put it in the header, but it looked to me like grant_type was required in the body.